src/components/PopupWithForm.js

```diff
@@ -30,4 +33,8 @@ export default class PopupWithForm extends Popup {
     this._popupForm.reset();
     super.close();
   }
+
+  setSubmitAction(action) {
```

Comment:
You should create a separate class `PopupWithConfirmation` for rendering a popup with confirmation button

Reply:
done